utils.c

```diff
@@ -103,7 +103,8 @@ safe_openat (int dirfd, const char *pathname, int flags, mode_t mode)
 int
 file_exists_at (int dirfd, const char *pathname)
 {
-  return faccessat (dirfd, pathname, F_OK, AT_SYMLINK_NOFOLLOW|AT_EACCESS);
+  struct stat buf;
+  return fstatat(dirfd, pathname, &buf, AT_SYMLINK_NOFOLLOW);
```

Comment:
stat is slower than accessat.  Could we use it as a fallback instead?

```
int ret = faccessat ....;
if (ret < 0 && errno == EINVAL)
  {
    struct stat buf;
    return fstatat (dirfd, pathname, &buf, AT_SYMLINK_NOFOLLOW);
  }
return ret;
```

Reply:
Wasn't this just hangking on musl/alpine?

Reply:
@giuseppe I've added your fallback code, completed the Dockerfile and added an alpine test.

Reply:
@rhatdan do you mean it is slightly slower on alpine then?
Do you think a preprocessor instruction would be more appropriate to use one or the other variant?

Reply:
the patch LGTM, let's wait for @rhatdan to chime in as he had a comment